### PR TITLE
Manage Students: enable and disable all share settings

### PR DESCRIPTION
**Pull request by Erin007**

### CODE REVIEW

**Comment by Erin007** [comment]
@caleybrock Things are getting interesting... This code works to enable or disable all of the share settings for a section. But, I'm using the `updateShareSetting` function from `teacherSectionsRedux`, not `manageStudentsRedux`, which means when the successful ajax request completes there isn't a state/props change in the `manageStudentsTable` that tells it to re-render. Updates to the share settings via enable or disable all are only shown on refresh. 

As I see it, there are 3 options: 
1.) Force a re-render when enable all or disable all is clicked
2.) Update some kind of state in the manageStudentsTable that knows when enable or disable worked
3.) Re-write updateShareSetting in the manageStudentsRedux.

Thoughts?

File: apps/src/templates/teacherDashboard/sections.js [hunk 1]
```diff
@@ -110,7 +110,9 @@ export function renderSectionTable(sectionId, loginType, courseId) {
     );
     ReactDOM.render(
       <Provider store={store}>
-        <ManageStudentsTable />
+        <ManageStudentsTable
+          sectionId={sectionId}
```
Comment (caleybrock) [inline apps/src/templates/teacherDashboard/sections.js]: sectionId is passed in via redux on line 87 above so you shouldn't need to pass it in here, just get it via redux.

**Comment by caleybrock** [comment]
Followed up via slack, but I think option 3 (code duplication until we launch the experiment) is best.

File: apps/src/templates/manageStudents/manageStudentsRedux.js [hunk 2]
```diff
@@ -108,6 +114,29 @@ export const addStudentsFailure = (numStudents, error, studentIds) => (
...
+  dispatch({type: UPDATE_SHARING_REQUEST});
+  return new Promise((resolve, reject) => {
+    $.ajax({
+      url: `dashboardapi/sections/${sectionId}/update_sharing_disabled`,
+      method: 'POST',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify({sharing_disabled: shareSetting}),
+    }).done(result => {
+      dispatch({
+        type: UPDATE_SHARING_SUCCESS,
```
Comment (Erin007) [inline apps/src/templates/manageStudents/manageStudentsRedux.js:127]: The part that's getting tricky is figuring out how  `UPDATE_SHARING_SUCCESS`, `UPDATE_SHARING_FAILURE` and `UPDATE_SHARING_REQUEST` fit in.  I've currently just copied them over from `teacherSectionsRedux`.  But the other ajax requests in `manageStudentsRedux` just do something like 
`  }).done((data) => {
    onComplete(null, data);
  }).fail((jqXhr, status) => {
    onComplete(status, null);
  });`

Also I think what we probably want to do is mirror something similar to success or failure of `EDIT_ALL` ?

**Comment by Erin007** [comment]
Closing in favor of #21413.  This approach would've required us to build a new API that separates editing and saving of the share settings. Instead we're extending the manageStudentsRedux patterns to create a new action.

**Pull request closed without merging**